repository/repository-core/src/main/java/org/eclipse/vorto/repository/core/indexing/IndexedModelInfo.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.vorto.repository.core.indexing;
+
+import java.util.Map;
+import org.eclipse.vorto.model.ModelId;
+import org.eclipse.vorto.model.ModelType;
+import org.elasticsearch.search.SearchHit;
```

Comment:
Hi @erlemantos 
Your model class contains a technical dependency to Elasticsearch. That way your interface IIndexService is bound to it as well. I would suggest to remove this and replace it a neutral representation. Your Elastic Service Implementation would then map the neutral and vice versa.

Reply:
The SearchHit is there as part of a convenience function, not as part of the model. It can be moved in a utils somewhere. A factory is overkill, I think.

Reply:
Well designing it like you have currently, would make your API and model classes depend on elastic search. 
As a proposal:  I would create a separate maven module, lets say "repository-search-elastic" , which is the only module having a dependency to elastic search classes. 

The repository-core module, would merely define the API which is used by other core components and is completely independent of elastic search. 
At configuration time, you would bind the repository with the elastic search component together.

Reply:
We can definitely add that as an improvement.